collection/skipmap/skipmap.go

```diff
@@ -332,6 +337,16 @@ func (s *Int64Map) LoadOrStore(key int64, value interface{}) (actual interface{}
 			valid                = true
 			pred, succ, prevPred *int64Node
 		)
+		if level == 0 {
+			level = s.randomlevel()
+			if level > hl {
+				// If the highest level is updated, usually means that many goroutines
+				// are inserting items. Hopefully we can find a better path in next loop.
+				// TODO(zyh): consider filling the preds if s.header[level].next == nil,
+				// but this strategy's performance is almost the same as the existing method.
+				continue
```

Comment:
> Hopefully we can find a better path in the next loop.

Does the path mean "search path" like searching an element in skiplist?

Why is the path of the next loop (more likely, or always) to be better?

Reply:
> Does the path mean "search path" like searching an element in skiplist?

Yes.

> Why is the path of the next loop (more likely, or always) to be better?

Because the frequency of the highest-level update is very low. If a new item is inserted at a new highest level, its previous node in the highest layer is always the header of the skip list(`s.header`). If multiple goroutines are updating the highest-level at this time, the `s.header` could be a hot spot.

If we continue the loop, the `findNode` function can use a new highest level to find the path, and the new path usually includes the new items in the new highest level. This strategy benefits the overall performance.

Reply:
To be honest, I thought filling the `preds` with `s.header` is better(we don't need to find a new path), but I don't see any performance change compared to the current one.